CMakeLists.txt

```
@@ -40,9 +40,7 @@ option(GPERF "Build with GPerfTools profiler" OFF)
 cmake_dependent_option(GPERF_HEAP_FIRST_GAME_ITERATION "Save heap profile of the first game iteration" OFF "GPERF" OFF)
 option(DISABLE_LTO "Disable link-time optimization (by default enabled in release mode)" OFF)
 option(PIE "Generate position-independent code" OFF)
-option(DIST "Dynamically link only glibc and SDL2" OFF)
-option(BINARY_RELEASE "Enable options for binary release" OFF)
-option(NIGHTLY_BUILD "Enable options for nightly build" OFF)
+option(DIST "Create app bundle for MacOSX/iOS" OFF)

```

Should this perhaps be ON by default?
Also, is it used only in 1 place? How about removing this as an option entirely?

---

It only applies for macOS, but requires sudo ... maybe we can switch it to a CPACK check instead?

---

Does it make sense to check CPACK there when the script sets CPACK just two lines after?

https://github.com/StephenCWills/devilutionX/blob/f102b766d8389f729db9ab2aca99e94d2142f707/CMakeLists.txt#L970

---

> It only applies for macOS, but requires sudo...

Actually, can you elaborate on this a bit? That code appears to run for both macOS and iOS. Does it simply do nothing for iOS?

Also, since it requires sudo, do we want to leave it as an option? Based on the documentation, it looks like this command makes the bundle standalone and portable to other devices. Maybe we can just rename it.

```suggestion
option(MACOSX_STANDALONE_APP_BUNDLE "Generate a portable app bundle to use on other devices (requires sudo)" OFF)
```

---

That's probably a good place to start :) Lets, make that change and then merge this.

---

I renamed the variable and double-checked MacOSX.yml. It wasn't using `BINARY_RELEASE` or `CMAKE_BUILD_TYPE` but rather setting the `BUILD_TYPE` environment variable so I missed it the first time. Also, it occurs to me that maybe `-DCPACK=ON` is unnecessary since it is defined as a `RELEASE_OPTION()`.